mlir/lib/Dialect/Vector/Transforms/VectorEmulateNarrowType.cpp

```diff
@@ -129,6 +132,45 @@ static FailureOr<Operation *> getCompressedMaskOp(OpBuilder &rewriter,
       auto denseAttr = DenseElementsAttr::get(newMaskType, newMaskValues);
       newMask = rewriter.create<arith::ConstantOp>(loc, newMaskType, denseAttr);
     }
+  } else if (constantOp) {
+    assert(shape.size() == 1 && "expected 1-D mask");
```

Comment:
Why do we have the assertion in this branch, but not others?

Reply:
@hanhanW Because it other branch it can handle multiple dimensional masks.

But should we simply only deal with 1d vectors in this pass?

Reply:
In this case, we can `return failure()` and add a TODO comment saying that it is not yet implemented. We should not crash in debug build and I believe the patterns can recover the state for this case.